scripts/zones/Mount_Zhayolm/npcs/_1p2.lua

```diff
@@ -0,0 +1,29 @@
+-----------------------------------
+-- Area: Mount Zhayolm
+--  NPC: Gates of Halvung
+-- !pos 189.000 -19.000 -20.000
+-----------------------------------
+local ID = require("scripts/zones/Mount_Zhayolm/IDs")
+-----------------------------------
+local entity = {}
+
+entity.onTrade = function(player, npc, trade)
+end
+
+entity.onTrigger = function(player, npc)
+    if player:hasKeyItem(xi.ki.CAST_METAL_PLATE) then
+        player:messageSpecial(ID.text.OPEN_GATES, xi.ki.CAST_METAL_PLATE)
+        npc:openDoor(15)
+    else
+        player:messageSpecial(ID.text.GATES_OF_HALVUNG)
+        player:setLocalVar("CheckedHalvungGate", 1)
```

Comment:
If we use local var, players will lose this var if they zone out and back in. 
I think we'd want to use a char var, no? Or is this proper behaviour for this quest?

Reply:
gross. Looks good!